File: android_samples_java/19.2 SukritiTest/app/src/main/java/com/sukriti/sukrititest/TwoFragment.java
package com.sukriti.sukrititest;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatButton;
import androidx.appcompat.widget.AppCompatEditText;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

public class TwoFragment extends Fragment {

    AppCompatEditText resultvaluetxt;
    AppCompatButton submitbtn;

    CallbackInterface callbackInterface;
    int n;

    @Override
    public void onAttach(@NonNull Activity activity) {
        super.onAttach(activity);
        callbackInterface = (SecondActivity) activity;
        n = getArguments().getInt("initvalue");
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
    {
        View view = inflater.inflate(R.layout.fragment_two, container, false);

        resultvaluetxt = view.findViewById(R.id.resultvaluetxt);
        submitbtn = view.findViewById(R.id.submitbtn);

        resultvaluetxt.setText(String.valueOf(n));
        submitbtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent();
                intent.putExtra("n",n);
                getActivity().setResult(1004, intent);
                getActivity().finish();
            }
        });

        return view;
    }

    public void incrementNumber(){
        if(n < 100) {
            n++;
            resultvaluetxt.setText(String.valueOf(n));
        }
    }

    public void decrementNumber(){
        if(n > 0) {
            n--;
            resultvaluetxt.setText(String.valueOf(n));
        }
    }
}

File: android_jetpack/SukritiTest/app/src/main/java/com/sukriti/sukrititest/TwoFragment.kt
package com.sukriti.sukrititest

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatButton
import androidx.appcompat.widget.AppCompatEditText
import androidx.fragment.app.Fragment
import com.sukriti.sukrititest.databinding.FragmentOneBinding
import com.sukriti.sukrititest.databinding.FragmentTwoBinding

class TwoFragment : Fragment() {
    private var _binding: FragmentTwoBinding? = null
    private val binding get() = _binding!!
    var callbackInterface: CallbackInterface? = null
    var n: Int = 0

    override fun onAttach(activity: Activity) {
        super.onAttach(activity)
        callbackInterface = activity as SecondActivity
        n = requireArguments().getInt("initvalue")
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentTwoBinding.inflate(inflater, container, false)

        binding.resultvaluetxt!!.setText(n.toString())
        binding.submitbtn!!.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                val intent = Intent()
                intent.putExtra("n", n)
                requireActivity().setResult(1004, intent)
                requireActivity().finish()
            }
        })

        return binding.root
    }

    fun incrementNumber() {
        if (n < 100) {
            n++
            binding.resultvaluetxt!!.setText(n.toString())
        }
    }

    fun decrementNumber() {
        if (n > 0) {
            n--
            binding.resultvaluetxt!!.setText(n.toString())
        }
    }
}


